### ISSUE TITLE
[BUG] AttributeError: 'PosixPath' object has no attribute 'endswith'

### ISSUE THREAD

**Issue by lorenzobalzani** [truera/trulens#1316]
**Bug Description**
I get this error:

```bash
AttributeError: 'PosixPath' object has no attribute 'endswith'
```

**To Reproduce**
```py
from trulens_eval import Feedback
```

**Expected behavior**
The error should not happen.

**Relevant Logs/Tracebacks**
```bash
2024-07-31 14:42:51.823 Uncaught app exception
Traceback (most recent call last):
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 589, in _run_script
    exec(code, module.__dict__)
  File "/path/to/entrypoint.py", line 69, in <module>
    main()
  File "/path/to/entrypoint.py", line 64, in main
    pg.run()
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/streamlit/navigation/page.py", line 286, in run
    exec(code, module.__dict__)
  File "/path/to/routes/chat.py", line 22, in <module>
    from promdc.backend.pipeline import Pipeline
  File "/path/to/pipeline.py", line 17, in <module>
    from .evaluation.evaluation import Evaluator
  File "/path/to/evaluation.py", line 1, in <module>
    from trulens_eval import Feedback
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/trulens_eval/__init__.py", line 16, in <module>
    from trulens_eval.utils.imports import check_imports
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/trulens_eval/utils/imports.py", line 20, in <module>
    from pip._internal.req import parse_requirements
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1138, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1078, in _find_spec
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/_distutils_hack/__init__.py", line 90, in find_spec
    return method()
           ^^^^^^^^
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/_distutils_hack/__init__.py", line 138, in spec_for_pip
    if sys.version_info >= (3, 12) or self.pip_imported_during_build():
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/_distutils_hack/__init__.py", line 150, in pip_imported_during_build
    return any(
           ^^^^
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/_distutils_hack/__init__.py", line 151, in <genexpr>
    cls.frame_file_is_setup(frame) for frame, line in traceback.walk_stack(None)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/to/envs/prometeia/lib/python3.11/site-packages/_distutils_hack/__init__.py", line 160, in frame_file_is_setup
    return frame.f_globals.get('__file__', '').endswith('setup.py')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'PosixPath' object has no attribute 'endswith'
```

**Environment:**
 - OS: Mac OS, Apple M3
 - Python Version: 3.12
 - TruLens version: git+https://github.com/truera/trulens@v0.33.0#subdirectory=trulens_eval
 - Versions of other relevant installed libraries:
```bash
streamlit~=1.36.0
langchain~=0.2.5
langchain-openai~=0.1.7
langchain-community~=0.2.0
loguru~=0.7.2
pillow~=10.3.0
pypdf~=3.17.0
tiktoken~=0.7.0
azure-ai-formrecognizer~=3.3.2
pandas~=2.1.2
qdrant-client~=1.8.0
openai~=1.30.0
tabulate~=0.9.0
streamlit-datalist~=0.0.5
streamlit-extras~=0.4.0
streamlit_antd_components~=0.3.2
plotly~=5.22.0
natsort~=8.4.0
trulens-eval-0.33.0 @ git+https://github.com/truera/trulens@v0.33.0#subdirectory=trulens_eval
numpy~=1.26.4
```

**Context**
I am running a Streamlit environment.

@sfc-gh-chu @sfc-gh-jreini

**Comment by lorenzobalzani** [comment]
I do think that this LLM response does not resolve the issue @sfc-gh-dhuang.

**Comment by sfc-gh-pdharmana** [comment]
Hey @lorenzobalzani I tried to reproduce error locally and could not do it. One question: you mentioned python version as 3.12 but the error trace mentions the version as 3.11. Could you please check the environment?

**Comment by lorenzobalzani** [comment]
> Hey @lorenzobalzani I tried to reproduce error locally and could not do it. One question: you mentioned python version as 3.12 but the error trace mentions the version as 3.11. Could you please check the environment?

That's my bad, I've figured out and forgot to update the post. I solved by using an environment with >= Python 3.12.
